rbd/admin/admin.go

```diff
@@ -49,3 +49,32 @@ func NewLevelSpec(pool, namespace, image string) LevelSpec {
 func NewRawLevelSpec(spec string) LevelSpec {
 	return LevelSpec{spec}
 }
+
+// ImageSpec values are used to identify RBD image wherever Ceph APIs
+// require an image_spec/image_id_spec using image name/id and optional
+// pool and namespace.
+type ImageSpec struct {
+	spec string
+}
+
+// NewImageSpec is used to construct a ImageSpec given a image name/id
+// and optional namespace and pool names.
+func NewImageSpec(pool, namespace, image string) ImageSpec {
+	var s string
+	if pool != "" && namespace != "" {
```

Comment:
I think we had discussed adding a comment explaining that this code isn't the same as the LevelSpec code despite it looking superficially similar. Please do so.